### PR TITLE
#T-218: country api + dashboard editor

### PR DESCRIPTION
**Pull request by michalpulpan**
👋 
added country api (@VL-CZ with generic views! 😂 ) and dashboard editor 😊 

https://github.com/ecoseller/ecoseller/assets/34132752/f0775089-b5da-48b6-832e-69efbebafbb9

### CODE REVIEW

**Comment by VL-CZ** [comment]
@michalpulpan 
You know, I'm something of generic view lover myself 😄

File: src/dashboard/pages/api/country/[code].tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+};
+
+const handler = async (req: NextApiRequest, res: NextApiResponse) => {
+  /**
+   * This is a wrapper for the country detail api in the backend
+   */
+  // get the cart data from the backend
+  const method = req.method as HTTPMETHOD;
+
+  if (method === "PUT") {
```
Comment (lakatop) [inline src/dashboard/pages/api/country/[code].tsx:72]: Maybe I am missing something, but is it necessary to do swith here, when there is already one in the countryDetailAPI method?

File: src/dashboard/pages/api/country/index.tsx [hunk 2]
```diff
@@ -37,13 +54,21 @@ export const countryListAPI = async (
...
-   * It returns whole cart data from the backend
+   * This is a wrapper for the country list api in the backend
    */
   // get the cart data from the backend
-  return countryListAPI("GET", req, res)
-    .then((data) => res.status(200).json(data))
-    .catch((error) => res.status(400).json(null));
+  const method = req.method as HTTPMETHOD;
+
+  if (method === "POST") {
```
Comment (lakatop) [inline src/dashboard/pages/api/country/index.tsx:62]: Same here.

File: src/dashboard/pages/api/product/price-list/index.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+};
+
+const handler = async (req: NextApiRequest, res: NextApiResponse) => {
+  /**
+   * This is a wrapper for the pricelist list api in the backend
+   */
+  // get the cart data from the backend
+  const method = req.method as HTTPMETHOD;
+
+  if (method === "POST") {
```
Comment (lakatop) [inline src/dashboard/pages/api/product/price-list/index.ts:59]: Same.

**Review by lakatop — CHANGES_REQUESTED** [review]

**Review by lakatop — APPROVED** [review]

**Pull request merged**